realm/realm-library/src/main/cpp/io_realm_internal_OsSet.cpp

```diff
@@ -51,6 +51,55 @@ Java_io_realm_internal_OsSet_nativeCreate(JNIEnv* env, jclass, jlong shared_real
     return reinterpret_cast<jlong>(nullptr);
 }
 
+JNIEXPORT jboolean JNICALL
+Java_io_realm_internal_OsSet_nativeIsValid(JNIEnv* env, jclass, jlong set_ptr) {
+    try {
+        auto& set = *reinterpret_cast<realm::object_store::Set*>(set_ptr);
+        return set.is_valid();
+    }
+    CATCH_STD()
+    return false;
+}
+
+JNIEXPORT jobject JNICALL
+Java_io_realm_internal_OsSet_nativeGetValueAtIndex(JNIEnv* env, jclass, jlong set_ptr, jint position) {
+    try {
+        auto& set = *reinterpret_cast<realm::object_store::Set*>(set_ptr);
+        const Mixed& value = set.get_any(position);
+        if (value.is_null()) {
+            return nullptr;
+        } else {
+            const DataType& type = value.get_type();
```

Comment:
Seems like something that would be somewhere else. Maybe abstract into a utility function.

Reply:
@clementetb and I will go through the native part of things at some point. There is a lot of boiler plate in this class which can for sure be eliminated or at least refactored - this is mainly due to me being so "proficient" in C++